SPECS/rook/rook.spec

```diff
@@ -0,0 +1,1192 @@
+csi_prov_image=$(sed -ne "s|.*DefaultProvisionerImage.*= \"\(.*provisioner:.*\)\"|\1|p" %{spec_go})
+csi_attach_image=$(sed -ne "s|.*DefaultAttacherImage.* \"\(.*attacher:.*\)\"|\1|p" %{spec_go})
+csi_snap_image=$(sed -ne "s|.*DefaultSnapshotterImage.* \"\(.*snapshotter:.*\)\"|\1|p" %{spec_go})
+csi_resize_image=$(sed -ne "s|.*DefaultResizerImage.* \"\(.*resizer:.*\)\"|\1|p" %{spec_go})
+#  16.2.4.26 is replaced at build time by _service
+#  rook_container_version is updated by update-tarball.sh:
+%global rook_container_version 1.6.2
+%global rook_image       %{registry}/rook/ceph:%{rook_container_version}
+%global ceph_image       %{registry}/ceph/ceph:16.2.4.26
+%global ceph_csi_image   $ceph_csi_image
+
+# remove symbols unsupported by k8s (+) from version
+version_full=%{version}
+version_noplus="${version_full//[+]/_}"
+%global version_parsed "${version_noplus}-%{release}"
+
+linker_flags=(
+    # Set Rook version - absolutely required
+    "-X" "github.com/rook/rook/pkg/version.Version=%{version_parsed}"
+    "-X" "github.com/rook/rook/pkg/operator/ceph/csi.DefaultCSIPluginImage=${ceph_csi_image}"
+    "-X" "github.com/rook/rook/pkg/operator/ceph/csi.DefaultRegistrarImage=${csi_reg_image}"
+    "-X" "github.com/rook/rook/pkg/operator/ceph/csi.DefaultProvisionerImage=${csi_prov_image}"
+    "-X" "github.com/rook/rook/pkg/operator/ceph/csi.DefaultAttacherImage=${csi_attach_image}"
+    "-X" "github.com/rook/rook/pkg/operator/ceph/csi.DefaultSnapshotterImage=${csi_snap_image}"
+    "-X" "github.com/rook/rook/pkg/operator/ceph/csi.DefaultResizerImage=${csi_resize_image}"
+)
+build_flags=("-ldflags" "${linker_flags[*]}" "-mod=vendor" "-v" "-a")
+
+for cmd in cmd/* ; do
+  go build "${build_flags[@]}" -o $(basename $cmd) ./$cmd
+done
+
+# BUILDING TEST BINARIES, NOTES:
+# Building test binaries works differently than main binaries; test binaries are built by 'go test',
+#     not 'go build' or 'go install'.
+# Spec build tooling provides 'gotest', but it expects to run the tests, which we cannot do with
+#     the integration tests at build time, so we run this manually. This may be fragile.
+# To compile but not run test binaries, we don't need the build flags needed by main binaries, but
+#     we do need: -c (compile test binary) and -o=<output-location> (output file)
+# 'goprep' does not set GOTPATH or GOBIN despite what the documentation might say; that is set in
+#     'gobuild', so we need to set it for our manual run of 'go test'.
+# Because this is a test binary which we SHOULD NOT ship to customers, we shouldn't need to follow
+#     every single go build best practice, and we don't need to worry about this becoming too out of
+#     date. We can specify some important flags for debugging bad builds:
+#         -v (orint package names), -x (print commands)]
+#    and flags to get rid of RPMLINT report warnings/errors:
+#         -buildmode=pie (position-independent executable)
+GOPATH=%{_builddir}/go GOBIN="${GOPATH}"/bin \
```

Comment:
Could you surround test-only part with `%if %{with_check}...%endif`? This makes non-test builds faster.

Reply:
Actually this seems to be both building and running the tests. In that case, let's just move all of it into a `%check` section. You'd need to create it for this spec, I think it's not present. The `%check` section doesn't need to be surrounded by `%9f {with_check}`.

Reply:
I believe I addressed the comment but I will hold off resolving in case this is not exactly what you meant.

Reply:
Looks good, just move the comment inside `%check` as well, so it's clear the comment is part of whatever is happening inside that section.

Reply:
removed the test binaries entirely as I interacted with them extensively and never got them to work.